src/coreclr/tools/Common/Compiler/DependencyAnalysis/ObjectDataBuilder.cs

```diff
@@ -85,42 +86,37 @@ public void EmitByte(byte emit)
 
         public void EmitShort(short emit)
         {
-            EmitByte((byte)(emit & 0xFF));
-            EmitByte((byte)((emit >> 8) & 0xFF));
+            Span<byte> buffer = stackalloc byte[sizeof(short)];
+            BinaryPrimitives.WriteInt16LittleEndian(buffer, emit);
+            _data.Append(buffer);
```

Comment:
We could also do this a little bit more efficiently by asking the `ArrayBuilder` for buffer of certain size (sorta like `IBufferWriter.GetSpan`) and avoid the temporary stackalloc. Should we do that instead?

Reply:
You would need to measure it.